modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -275,17 +279,25 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
   def addSequenceToQueue(q: EventQueue, qid: QueueId, seqId: Observation.Id): IO[Either[SeqexecFailure, Unit]] =
     addSequencesToQueue(q, qid, List(seqId))
 
-  private def removeSeq(qid: QueueId, seqId: Observation.Id): Endo[EngineState] = st => (
-    for {
-      q <- st.queues.get(qid)
-      if (q.queue.contains(seqId))
-      if (q.status(st) =!= BatchExecState.Running || st.executionState.sequences.get(seqId)
-        .map(seq => !seq.status.isRunning && !seq.status.isCompleted).getOrElse(true))
-    } yield queueO(qid).modify(_.removeSeq(seqId))(st)
-  ).getOrElse(st)
+  private def removeSeq(qid: QueueId, seqId: Observation.Id): executeEngine.HandleType[Unit] =
+    executeEngine.get.flatMap{ st => (
+      for {
+        q <- st.queues.get(qid)
+        if (q.queue.contains(seqId))
+        sstOp = st.executionState.sequences.get(seqId).map(_.status)
+        if (q.status(st) =!= BatchExecState.Running ||
+          sstOp.map(sst => !sst.isRunning && !sst.isCompleted).getOrElse(true))
+      } yield executeEngine.modify(queueO(qid).modify(_.removeSeq(seqId))) *>
+        ((q.cmdState, q.status(st)) match {
+          case (_, BatchExecState.Completed)         => executeEngine.unit
+          case (BatchCommandState.Run(o, u, c), _)
+            if sstOp.map(_.isError).getOrElse(false) => runQueue(qid, o, u, c)
+          case _                                     => executeEngine.unit
+        })
+    ).getOrElse(executeEngine.unit)}
```

Comment:
If the removed sequence was in a failed state, it might have been stalling the queue, so try to run it.